Tests/ListAddMethodsTests.cs

```diff
@@ -0,0 +1,132 @@
+        {
+            var instance = _list.CreateInstance(sourceArray);
+            instance.AddFront(3);
+            Assert.AreEqual(instance.Count, expectedArray.Length);
+            CollectionAssert.AreEqual(expectedArray, instance);
+        }
+
+        private static readonly object[] AddByIndex_1 = new[] { new object[] { new int[] { 1, 2 }, 1, new int[] { 1, 3, 2 } } };
+        private static readonly object[] AddByIndex_2 = new[] { new object[] { new int[] { 1, 2 }, 2, new int[] { 1, 2, 3 } } };
+        [TestCaseSource("AddByIndex_1")]
+        [TestCaseSource("AddByIndex_2")]
+        [TestCase(new int[] { }, 0, new int[] { 3 })]
+        public void AddByIndex_WhenIndexIsNotLessZeroOrIndexMoreThenSize_WhenAny_ShouldAddFront(int[] sourceArray, int index, int[] expectedArray)
+        {
+            var instance = _list.CreateInstance(sourceArray);
+            instance.AddByIndex(index, 3);
+            Assert.AreEqual(instance.Count, expectedArray.Length);
+            CollectionAssert.AreEqual(expectedArray, instance);
+        }
+
+        [TestCase(-1)]
+        [TestCase(1)]
+        public void AddByIndex_WhenIndexLessZeroOfIndexMoreThenSize_ShouldThrowArgumentOutOfRangeException(int pos)
+        {
+            try
+            {
+                _list.AddByIndex(pos, 0);
+            }
+            catch (ArgumentOutOfRangeException ex)
+            {
+                Assert.AreEqual("Index is out of range", ex.ParamName);
+                Assert.Pass();
+            }
+
+            Assert.Fail();
+        }
+
+        private static readonly object[] AddMany_1 = new[] { new object[] { new int[] { 1, 2, 3 }, new List<int> { 4, 5 }, new int[] { 1, 2, 3, 4, 5 } } };
+        private static readonly object[] AddMany_2 = new[] { new object[] { new int[] { }, new List<int> { 1, 2, 3 }, new int[] { 1, 2, 3 } } };
+        [TestCaseSource("AddMany_1")]
+        [TestCaseSource("AddMany_2")]
+        public void AddMany_WhenAny_ShouldAddItemsToEnd(int[] sourceArray, IEnumerable<int> additionalList, int[] expectedArray)
+        {
+            var instance = _list.CreateInstance(sourceArray);
+            instance.Add(additionalList);
+            Assert.AreEqual(instance.Count, expectedArray.Length);
+            CollectionAssert.AreEqual(expectedArray, instance);
+        }
+
+        private static readonly object[] AddFrontMany_1 = new[] { new object[] { new int[] { 1, 2, 3 }, new List<int> { 4, 5 }, new int[] { 4, 5, 1, 2, 3 } } };
+        private static readonly object[] AddFrontMany_2 = new[] { new object[] { new int[0], new List<int> { 1, 2, 3 }, new int[] { 1, 2, 3 } } };
+        [TestCaseSource("AddFrontMany_1")]
+        [TestCaseSource("AddFrontMany_2")]
+        public void AddManyFront_WhenAny_ShouldAddItemsToFront(int[] sourceArray, IEnumerable<int> additionalList, int[] expectedArray)
+        {
+            var instance = _list.CreateInstance(sourceArray);
```

Comment:
Split all test body on three parts as I explained Arrange, Act, Assert

Reply:
by region or by name of method?